assets/js/components/Dashboard/DatePicker.tsx

```diff
@@ -0,0 +1,68 @@
+import moment from "moment";
+import React, { useRef, useState } from "react";
+import { Form, Overlay, Popover } from "react-bootstrap";
+import Calendar from "react-calendar";
+
+interface DatePickerProps {
+  selectedDate: string;
+  onChange: (date: string) => void;
+  minDateString?: string;
+  maxDateString?: string;
+}
+
+const DatePicker = ({
+  selectedDate,
+  onChange,
+  minDateString,
+  maxDateString,
+}: DatePickerProps) => {
+  const [showOverlay, setShowOverlay] = useState(false);
+  const ref = useRef(null);
+  const minDate = minDateString ? new Date(minDateString) : undefined;
+  const maxDate = maxDateString ? new Date(maxDateString) : undefined;
```

Comment:
**quibble (non-blocking):** i like to avoid `undefined` as a programmer-set value and would suggest using `null` unless something _absolutely needs_ an `undefined` value

Reply:
I normally agree, but this was due to the accepted props in the `Calendar` component. Only `Date` objects are accepted and unfortunately `new Date(undefined)` returns a valid object.